MetaMask/metamask-extension
# add popover onClickOutside

## **Description**
Add additional functionality to Popover component to support an onClickOutside callback to give the dev more flexibility on the state of when the Popover should be opened and closed.

## **Related issues**

Fixes: #21732 

## **Manual testing steps**

1. Run localhost of this PR
2. Visit http://localhost:6006/?path=/story/components-componentlibrary-popover--on-click-outside
3. Click to open Popover
4. Click Popover to confirm it DOES **NOT** close
5. Click somewhere outside of the Popover to confirm it DOES close

## **Screenshots/Recordings**



### **Before**
n/a


### **After**

https://github.com/MetaMask/metamask-extension/assets/26469696/7ffff799-e7c8-4ae6-9cfc-5fa9a2916451




## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
